.golangci.yml

```diff
@@ -0,0 +1,117 @@
+run:
+  modules-download-mode: readonly
+  issues-exit-code: 0
+
+linters-settings:
+  exhaustive:
+    default-signifies-exhaustive: false
+  gci:
+    sections:
+      - standard
+      - default
+      - prefix(github.com/opensearch-project/opensearch-go)
+  goconst:
+    min-occurrences: 5
+    ignore-tests: true
+  godot:
+    scope: toplevel
+    exclude:
+      - go-sumtype:decl
+      - check interfaces
+    capital: true
+  godox:
+    keywords:
+      - BUG
+      - FIXME
+      - HACK
+  goimports:
+    local-prefixes: github.com/opensearch-project/opensearch-go
+  gosimple:
+    go: "1.18"
+    checks: ["all"]
+  lll:
+    line-length: 130
+    tab-width: 4
+  nolintlint:
+    allow-unused: false
+    allow-leading-space: false
+    allow-no-explanation: []
+    require-explanation: true
+    require-specific: true
+  staticcheck:
+    go: "1.18"
+    checks: ["all"]
+  stylecheck:
+    go: "1.18"
+    checks: ["all"]
+    dot-import-whitelist: []
+    initialisms: []
+    http-status-code-whitelist: []
+  unparam:
+    check-exported: true
+  unused:
+    go: "1.18"
+  whitespace:
+    multi-if: false
+    multi-func: false
+  tagliatelle:
+
+linters:
+  enable-all: true
+  disable:
```

Comment:
Are we keeping all linters disabled in this PR?

Reply:
Not really. We enabled all linters but disabled the ones that return warnings. In the next PRs I'm going to fix them and enable one by one

Reply:
Oh my bad, can you also create an issue with the list of linters to be enabled may be on a checklist? It would be easier for the upcoming PRs.